lib/internal/test_runner/runner.js

```diff
@@ -377,20 +386,36 @@ function run(options) {
   if (setup != null) {
     validateFunction(setup, 'options.setup');
   }
+  if (testNamePatterns != null) {
+    if (!ArrayIsArray(testNamePatterns)) {
+      testNamePatterns = [testNamePatterns];
+    }
+    validateArray(testNamePatterns, 'options.testNamePatterns');
+    testNamePatterns = ArrayPrototypeMap(testNamePatterns, (value, i) => {
+      if (value instanceof RegExp) {
```

Comment:
this isn't a robust regexp check if user code has been able to modify `RegExp`; you may need something akin to https://github.com/inspect-js/is-regex/blob/main/index.js to do this properly.

Reply:
@ljharb I see there is a `isRegExp` function in `node:util/types`, but if I understand correctly, it seems to be deprecated. 
Can I use that directly? (And out of curiosity, why has it been deprecated?)

Reply:
I'm not sure why it's deprecated, but it certainly seems like something you could use directly off of `const types = require('internal/util/types');`?

Reply:
Yeah, my question was mainly if it is a "valid" check, and if that would be a good solution for the concern regarding RegExp being modified :)
BTW, I tried finding the source of `isRegExp` but I was unable too, if anyone can share it, that would be an awesome way for me to learn more 🙂

Reply:
@atlowChemi in the Node.js parts (where it gets to V8) or how V8's value::IsRegExp works?

Reply:
Thanks @benjamingr  `value::IsRegExp` helped me find what I was looking for 🙂